implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/DynatraceConfig.java

```diff
@@ -76,8 +85,37 @@ default DynatraceApiVersion apiVersion() {
                 .orElse(DynatraceApiVersion.V1);
     }
 
+    default String metricKeyPrefix() {
+        // returns an empty string if nothing is set in the properties.
+        return getString(this, "metricKeyPrefix").orElse("");
+    }
+
+    @Nullable
+    default Map<String, String> defaultDimensions() {
+        return Collections.emptyMap();
+    }
+
+    default Boolean enrichWithOneAgentMetadata() {
+        // defaults to false if nothing is set.
+        return getBoolean(this, "enrichWithOneAgentMetadata")
+                .orElse(false);
+    }
+
     @Override
     default Validated<?> validate() {
+        Validated<DynatraceApiVersion> checkApiVersion = checkRequired("apiVersion", DynatraceConfig::apiVersion).apply(this);
```

Comment:
```suggestion
        // verify if the specified apiVersion corresponds to a value in the DynatraceApiVersion enum
        Validated<DynatraceApiVersion> checkApiVersion = checkRequired("apiVersion", DynatraceConfig::apiVersion).apply(this);
```
Otherwise one might think that this is always passing because of the v1 fallback.